### ISSUE TITLE
Allow an independent start date for a task that depends on another task

### ISSUE THREAD

**Issue by TheresaTeng** [UoMResearchIT/CapX#143]
When task B depends on task A's completion, task A can be set as task B's predecessor. However task B doesn't necessarily start immediately after A has been completed, it may start later. In this case it would be useful for the system to allow user setting a start date for task B, regardless of task A's finish date.

**Comment by aharwood2** [comment]
I think the way I will implement this will be to add a separate field when a predecessor is selected called "Lag" which can be set in days to lags the start of the task by. That way the start date cannot be considered a "fixed start" set manually but instead is still a driven quantity computed by the scheduler -- just one based on a couple of different input parameters. This mimics the behaviour of MSP which implements it the same way through a S-F constraint.

**Comment by PhilBradbury** [comment]
I came here to add this to the board, but it's here already. :)

Lag is a great way of describing it - I have tasks A and B and want B to start "an amount of time" after task A but to move automatically with Task A if that is resheduled. I try to link all my tasks so that they all move if a date is changed.

**Comment by aharwood2** [comment]
- [x] Add fixed lag to DB
- [x] Add validation to ensure it is a integer greater than -1
- [x] Adjust scheduler code to take it into account automatically when end date of a predecessor is set
- [x] Adjust disabled logic with info icons so that only enabled when predecessor selected
